### PR TITLE
Update node-feature-discovery deployment

### PR DESCRIPTION
**Pull request by e0ne**
1. Use helm chart updated from NFD repo
2. network-operator specifies the latest build of the NFD
3. nfd-master '--instance' option is set to 'nvidia.networking' be default

### CODE REVIEW

File: deployment/network-operator/charts/node-feature-discovery/templates/master.yaml [hunk 1]
```diff
@@ -58,17 +58,19 @@ spec:
                 fieldPath: spec.nodeName
           command:
             - "nfd-master"
-          
           resources:
             {{- toYaml .Values.master.resources | nindent 12 }}
+          args:
+            { { - if .Values.master.instance | empty | not } }
+            - "--instance={{ .Values.master.instance }}"
```
Comment (moshe010) [inline deployment/network-operator/charts/node-feature-discovery/templates/master.yaml:65]: I don't think we want it to be configurable. you can hardcode the value nvidia.networking or something else

File: deployment/network-operator/charts/node-feature-discovery/templates/nfd-worker-conf.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{{/*
+   Copyright 2020 NVIDIA
```
Comment (moshe010) [inline deployment/network-operator/charts/node-feature-discovery/templates/nfd-worker-conf.yaml:2]: s/2020/2021

File: deployment/network-operator/values.yaml [hunk 3]
```diff
@@ -24,6 +24,11 @@ sriovNetworkOperator:
 
 # Node Feature discovery chart related values
 node-feature-discovery:
+  image:
+    repository: gcr.io/k8s-staging-nfd/node-feature-discovery
+    tag: v0.6.0-233-g3e00bfb
```
Comment (moshe010) [inline deployment/network-operator/values.yaml:29]: can we do it deafult in the app vesion

File: deployment/network-operator/values.yaml [hunk 4]
```diff
@@ -24,6 +24,11 @@ sriovNetworkOperator:
...
+  master:
+    instance: "network-operator"
```
Comment (moshe010) [inline deployment/network-operator/values.yaml:31]: let remove this to be hard coded

**Review by moshe010 — CHANGES_REQUESTED** [review]

File: deployment/network-operator/charts/node-feature-discovery/templates/clusterrole.yaml [hunk 5]
```diff
@@ -15,7 +15,7 @@
 */}}
 
 {{- if .Values.rbac.create }}
-apiVersion: rbac.authorization.k8s.io/{{ .Values.rbac.apiVersion }}
+apiVersion: rbac.authorization.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (e0ne) [reply deployment/network-operator/charts/node-feature-discovery/templates/clusterrole.yaml:18]: rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding is deprecated in v1.17+, unavailable in v1.22+, so only stable API is supported in NFD

File: deployment/network-operator/charts/node-feature-discovery/templates/master.yaml [hunk 6]
```diff
@@ -46,7 +46,7 @@ spec:
         - name: master
           securityContext:
             {{- toYaml .Values.master.securityContext | nindent 12 }}
-          image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
```
[the remark this answers is not in the document]
Comment (e0ne) [reply deployment/network-operator/charts/node-feature-discovery/templates/master.yaml:49]: Let's keep it as exists in NFD but hardcode version

File: deployment/network-operator/charts/node-feature-discovery/templates/master.yaml (hunk 1 above)
Comment (e0ne) [reply deployment/network-operator/charts/node-feature-discovery/templates/master.yaml:65]: Let's hardcode it in values

File: deployment/network-operator/values.yaml (hunk 4 above)
Comment (e0ne) [reply deployment/network-operator/values.yaml:31]: One we'll start using chart from NFD helm repository, we won't be able to hardcode it without network-operator and/or admission controller changes, so I)'m not sure that it's a good idea to do it

**Comment by AbdYsn** [comment]
logs can be found [here](http://13.74.249.42/nic_operator_helm-ci/93/). it seems the pod was not labeled with pci-15b3.present or something similar

File: deployment/network-operator/charts/node-feature-discovery/templates/worker.yaml [hunk 7]
```diff
@@ -39,30 +39,26 @@ spec:
...
+        resources:
+        {{- toYaml .Values.worker.resources | nindent 12 }}
         command:
         - "nfd-worker"
-        resources:
-          {{- toYaml .Values.worker.resources | nindent 12 }}
         args:
         - "--sleep-interval=60s"
         - "--server={{ include "node-feature-discovery.fullname" . }}-master:{{ .Values.master.service.port }}"
-        {{- with .Values.worker.options }}
```
Comment (e0ne) [inline deployment/network-operator/charts/node-feature-discovery/templates/worker.yaml:63]: It should be restored

File: deployment/network-operator/charts/node-feature-discovery/templates/master.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (adrianchiris) [reply deployment/network-operator/charts/node-feature-discovery/templates/master.yaml:49]: > Let's keep it as exists in NFD but hardcode version

+1 
ideally no changes to the chart itself but only override values in network-operator values.yaml file

File: deployment/network-operator/charts/node-feature-discovery/templates/nfd-worker-conf.yaml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Values.worker.configmapName }}
+  labels:
+  {{- include "node-feature-discovery.labels" . | nindent 4 }}
+data:
+  nfd-worker.conf: |
+  {{ .Values.worker.config | indent 4 }}
```
Comment (adrianchiris) [inline deployment/network-operator/charts/node-feature-discovery/templates/nfd-worker-conf.yaml:25]: we should actually use the configmap and  put our configs here and not under options WDYT ?

i.e override this value from the top level values.yaml in network-operator

File: deployment/network-operator/charts/node-feature-discovery/values.yaml [hunk 9]
```diff
@@ -17,8 +17,11 @@
 # Declare variables to be passed into your templates.
 
 image:
-  repository: quay.io/kubernetes_incubator/node-feature-discovery
-  pullPolicy: IfNotPresent
+  repository: gcr.io/k8s-staging-nfd/node-feature-discovery
+  # This should be set to 'IfNotPresent' for released version
+  pullPolicy: Always
```
Comment (adrianchiris) [inline deployment/network-operator/charts/node-feature-discovery/values.yaml:22]: change in top level values.yaml to IfNotPresent ?

**Review by adrianchiris — COMMENTED** [review]
@e0ne i expect the nfd chart itself to be identical to what was merged to NFD with the exception of version override in `node-feature-discovery/Chart.yaml`

is that the case ?

In addition

File: deployment/network-operator/charts/node-feature-discovery/templates/nfd-worker-conf.yaml (hunk 8 above)
Comment (e0ne) [reply deployment/network-operator/charts/node-feature-discovery/templates/nfd-worker-conf.yaml:25]: Good catch. Let me try it

File: deployment/network-operator/values.yaml [hunk 10]
```diff
@@ -34,15 +38,16 @@ node-feature-discovery:
...
+    config: |
       sources:
-        pci:
-          deviceClassWhitelist:
-            - "02"
-            - "0200"
-            - "0207"
-          deviceLabelFields:
-            - vendor
+          pci:
```
Comment (e0ne) [inline deployment/network-operator/values.yaml:43]: An indentation looks to big but it works as expected

**Comment by adrianchiris** [comment]
In regards to commit message:
This is actually the first time we use HELM chart from NFD, so commit message should reflect that.

File: deployment/network-operator/values.yaml (hunk 10 above)
Comment (moshe010) [reply deployment/network-operator/values.yaml:43]: @e0ne  please open an issue in github and in our RM for this. 
I will merged as it has it doesn't effect functionality

**Review by moshe010 — APPROVED** [review]

**Pull request merged**